<?xml version="1.0"?>
<launch>
    <param name="use_sim_time" value="true" />
    
    <!-- run dual-laser localisation -->
   <node name="ndt_mcl" pkg="ndt_localization" type="ndt_mcl_localization" output="screen">
      <!-- jff file with map -->
    <param name="map_file" value="$(find ndt_fuser)/maps/basement2d_map.jff"  />
    	<param name="sensor_pose_x" value="0.695" />
	<param name="sensor_pose_y" value="-0.01" />
	<param name="sensor_pose_z" value="0.0" />
	<param name="sensor_pose_t" value="-0.0069813" />
	<param name="initial_y" value="0.0" />
	<param name="initial_x" value="0.0" />
	<param name="initial_t" value="0.0" />
    <!-- resolution of jff map -->
    <param name="resolution" value="0.2" />
    <!-- show aprticles for debug purposes -->
    <param name="show_particles" value="true" />
    <!-- particle count for ndt-mcl -->
    <param name="particle_count" value="500" />
    <param name="show_pose" value="true"/>
    <param name="fraction" value="1.0"/>
    <param name="force_SIR" value="true" />
    <param name="input_topic_name" value="laserscan" />
    <param name="Laser" value="true" />
    <param name="odom_tf" value="world" />
    <param name="root_tf" value="world" />
    <param name="base_tf" value="odom_base_link" />
    <param name="init_type" value="ndt"/>
    <param name="opening" value="3.1415"/>

    <rosparam param="begins">[1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0, 5.5, 6.0, 6.5, 7.0, 7.5, 8.0, 8.5, 9.0, 9.5, 10.0, 10.5, 11.0, 11.5, 12.0, 12.5, 13.0, 13.5, 14.0, 14.5, 15.0, 15.5, 16.0, 16.5, 17.0, 17.5, 18.0, 18.5, 19.0, 19.5, 20.0, 20.5, 221.0, 21.5 ]</rosparam>
    <rosparam param="ends">[1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0, 5.5, 6.0, 6.5, 7.0, 7.5, 8.0, 8.5, 9.0, 9.5, 10.0, 10.5, 11.0, 11.5, 12.0, 12.5, 13.0, 13.5, 14.0, 14.5, 15.0, 15.5, 16.0, 16.5, 17.0, 17.5, 18.0, 18.5, 19.0, 19.5, 20.0, 20.5, 221.0, 21.5, 22.0]</rosparam>

 <!--
    <rosparam param="begins">[1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0, 18.0, 19.0, 20.0]</rosparam>
    <rosparam param="ends">[2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0, 18.0, 19.0, 20.0, 21.0]</rosparam>
 -->
 <!--
 <rosparam param="begins">[1.0, 3.0, 5.0, 7.0, 9.0, 11.0, 13.0, 15.0, 17.0, 19.0, 21.0]</rosparam>
 <rosparam param="ends">[3.0, 5.0, 7.0, 9.0, 11.0, 13.0, 15.0, 17.0, 19.0, 21.0, 23.0]</rosparam>
 -->
 <!--
  <rosparam param="begins">[1.0, 4.0, 7.0, 10.0, 13.0, 16.0, 19.0, 22.0]</rosparam>
   <rosparam param="ends">[4.0, 7.0, 10.0, 13.0, 16.0, 19.0, 22.0, 25.0]</rosparam>
 -->
<!--
   <rosparam param="begins">[1.0, 5.0, 9.0, 13.0, 17.0, 21.0]</rosparam>
   <rosparam param="ends">[5.0, 9.0, 13.0, 17.0, 21.0, 25.0]</rosparam>
-->
   </node>


  <node name="ndt_map_publisher" pkg="ndt_localization" type="ndt_map_publisher" output="screen">
    <param name="map_file" value="$(find ndt_fuser)/maps/basement2d_map.jff"  />
    <param name="resolution" value="0.2"/>
    <param name="map_rate" value="0.1"/>
    <param name="map_frame" value="world"/>
  </node>


  
</launch>
